File: java/07 - Demo 3 - Refactoring/src/test/java/extractmethod/PrintOwingTest.java
package extractmethod;

import static org.junit.Assert.assertEquals;

import java.io.ByteArrayOutputStream;
import java.io.OutputStream;
import java.io.PrintStream;

import org.junit.Test;

public class PrintOwingTest {

	private static final double AMOUNT = 8.0;
	private static final String NAME = "Name";

	@Test
	public void testPrintOwing() {
		OutputStream outputStream = new ByteArrayOutputStream();
		PrintStream printStream = new PrintStream(outputStream);
		System.setOut(printStream);
		
		Person person = new Person(NAME);
		
		person.printOwing(AMOUNT);
		
		assertEquals("Person has: name: " + NAME + ", amount: " + AMOUNT, outputStream.toString());
	}
}

File: csharp/07 - Demo 3 - Refactoring/Refactoring.Tests/ExtractMethod/PrintOwingTest.cs
using System;
using System.IO;
using System.Text;
using Xunit;
using FluentAssertions;
using Refactoring.ExtractMethod;

namespace Refactoring.Tests.ExtractMethod
{
    public class PrintOwingTest
    {
        private const double Amount = 8.0;
        private const string Name = "Name";

        
        [Fact]
        public void TestPrintOwing()
        {
            var memoryStream = new MemoryStream();
            StreamWriter sw = new StreamWriter(memoryStream);
            sw.AutoFlush = true;
            Console.SetOut(sw);
            
            var person = new Person(Name);

            person.PrintOwing(Amount);

            Encoding.ASCII.GetString(memoryStream.ToArray()).Should().Be($"Person has: name: {Name}, amount: {Amount}");
        }
    }
}